src/script/util/ephemeralValueStore.ts

```diff
@@ -27,11 +27,24 @@ import {Config} from '../auth/config';
  */
 let worker: ServiceWorker;
 
-export async function get(): Promise<any> {
+export async function getEphemeralValue(): Promise<any> {
   const worker = await getWorker();
   return sendMessage(worker, {action: 'get'});
 }
 
+export async function saveRandomEncryptionKey(): Promise<string> {
+  const secretKey = new Uint32Array(64);
+  window.crypto.getRandomValues(secretKey);
+  const hexKey: string[] = [];
+  for (var i = 0; i < secretKey.length; i++) {
+    const x = secretKey[i];
+    hexKey.push(`00${x.toString(16)}`.slice(-2));
```

Comment:
`.slice(-2)` always takes the last two characters from a string. So why do you add `00` in the beginning if you are not going to use it?

Reply:
Ah okay, I see. It's a fallback if the secretKey is empty I guess?
Anyway, let's use a `for...in` loop here.

Reply:
I just moved this piece of code to a different place. Haven't changed any existing business logic here.

Reply:
Where does it come from? I don't see it in the WebApp repository.

Reply:
All good. We turned it into a "for of" now.